commit 3ab6ca57de19ed1559accd632b46e27b1a4fd0eb
Author: Harald Sitter <sitter@kde.org>
Date:   Thu Dec 5 11:13:03 2013 +0100

    Do not crash when failing to load a theme, but instead exit.
    
    When failing to load the theme's qml file QDeclarativeView goes into error
    state, at which point we want to exit to prevent nullptr access problems.
    
    REVIEW: 114298

diff --git a/ksplash/ksplashqml/SplashApp.cpp b/ksplash/ksplashqml/SplashApp.cpp
index 7c528d0..05253ff 100644
--- a/ksplash/ksplashqml/SplashApp.cpp
+++ b/ksplash/ksplashqml/SplashApp.cpp
@@ -136,6 +136,12 @@ void SplashApp::screenGeometryChanged(int)
         }
         else {
             SplashWindow *w = new SplashWindow(m_testing);
+            // When running into a view error at this point it means that the
+            // theme file could not be loaded. We want to exit as otherwise we
+            // are exposed to nullptr crashes.
+            if (w->status() == QDeclarativeView::Error) {
+                ::exit(2);
+            }
             w->setGeometry(m_desktop->availableGeometry(i));
             w->setStage(m_stage);
             w->show();
@@ -146,4 +152,4 @@ void SplashApp::screenGeometryChanged(int)
     m_windows.erase(m_windows.begin() + i, m_windows.end());
 }
 
-#include "SplashApp.moc"
\ No newline at end of file
+#include "SplashApp.moc"
